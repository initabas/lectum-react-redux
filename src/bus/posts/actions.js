import types from './types';

export const postsActions = Object.freeze({
    fetchPosts: () => ({
        type:    types.FETCH_POSTS,
    }),
    fetchPostsSuccess: (posts) => ({
        type:    types.FETCH_POSTS_SUCCESS,
        payload: posts,
    }), 
    fetchPostsFail: (error) => ({
        type:    types.FETCH_POSTS_FAIL,
        payload: error,
        error:   true,
    }),
    
    createPost: (comment) => ({
        type:    types.CREATE_POST,
        payload: comment,
    }),
    createPostSuccess: (post) => ({
        type:    types.CREATE_POST_SUCCESS,
        payload: post,
    }), 
    createPostFail: (error) => ({
        type:    types.CREATE_POST_FAIL,
        payload: error,
        error:   true,
    }),

    deletePost: (id) => ({
        type:    types.DELETE_POST,
        payload: id,
    }),
    deletePostSuccess: (id) => ({
        type:    types.DELETE_POST_SUCCESS,
        payload: id,
    }), 
    deletePostFail: (error) => ({
        type:    types.DELETE_POST_FAIL,
        payload: error,
        error:   true,
    }),

    likePost: (postId) => ({
        type:    types.LIKE_POST,
        payload: postId,
    }),
    likePostSuccess: (likedPostIds) => ({
        type:    types.LIKE_POST_SUCCESS,
        payload: likedPostIds,
    }), 
    likePostFail: (error) => ({
        type:    types.LIKE_POST_FAIL,
        payload: error,
        error:   true,
    }),

    dislikePost: (postId) => ({
        type:    types.DISLIKE_POST,
        payload: postId,
    }),
    dislikePostSuccess: (dislikedPostIds) => ({
        type:    types.DISLIKE_POST_SUCCESS,
        payload: dislikedPostIds,
    }), 
    dislikePostFail: (error) => ({
        type:    types.DISLIKE_POST_FAIL,
        payload: error,
        error:   true,
    }),
});
