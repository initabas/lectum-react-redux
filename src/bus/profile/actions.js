import types from './types';

export const profileActions = Object.freeze({
    fillProfile: (profile) => ({
        type:    types.FILL_PROFILE,
        payload: profile,
    }),
});
