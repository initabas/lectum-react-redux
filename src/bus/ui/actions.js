import types from './types';

export const uiActions = Object.freeze({
    setFeedFetchingState: (state) => ({
        type:    types.SET_FEED_FETCHING_STATE,
        payload: state,
    }),
    initialize: (state) => ({
        type:    types.INITIALIZE,
    }),
    setOnLineStatus: (state) => ({
        type:    types.SET_ONLINE_STATUS,
        payload: state,
    }),
    setAuthFetchingState: (state) => ({
        type:    types.SET_AUTH_FETCHING_STATE,
        payload: state,
    }),
});
