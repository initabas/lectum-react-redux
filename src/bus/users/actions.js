import types from './types';

export const usersActions = Object.freeze({
    fetchUsers: () => ({
        type:    types.FETCH_USERS,
    }),
    fetchUsersSuccess: (users) => ({
        type:    types.FETCH_USERS_SUCCESS,
        payload: users,
    }), 
    fetchUsersFail: (error) => ({
        type:    types.FETCH_USERS_FAIL,
        payload: error,
        error:   true,
    }),
});
