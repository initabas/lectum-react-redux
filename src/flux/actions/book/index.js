// Instruments
import { CHANGE_PAGE } from './types';

export const changePage = (page) => ({
    type:    CHANGE_PAGE,
    payload: page,
});
