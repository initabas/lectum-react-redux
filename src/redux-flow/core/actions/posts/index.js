// Instruments
import { DELETE_POST, CREATE_POST, CREATE_POST_ASYNC } from './types';
import { lorem } from 'faker';

export const deletePost = (postId) => ({
    type:    DELETE_POST,
    payload: postId,
});

export const createPost = () => ({
    type:    CREATE_POST,
    payload: lorem.sentence(),
});

export const createPostAsync = () => async (dispatch) => {
    const response = await fetch('http://date.jsontest.com/');

    const { time } = await response.json();

    await new Promise((resolve) => setTimeout(resolve, 1000));

    dispatch({
        type:    CREATE_POST_ASYNC,
        payload: `A current time is: ${time}.`,
    });
};
