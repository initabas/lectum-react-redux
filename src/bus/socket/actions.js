import { socket } from 'init/socket';
import { uiActions } from '../ui/actions';

export const watchConnection = () => (dispatch) => {
    socket.on('connect', () => {
        dispatch(uiActions.setOnLineStatus(true));
    });
    socket.on('disconect', () => {
        dispatch(uiActions.setOnLineStatus(false));
    });
};

export const watchPosts = () => () => {
    // socket.on('connect', () => {
    //     dispatch(uiActions.setOnLineStatus(true));
    // });
    // socket.on('disconect', () => {
    //     dispatch(uiActions.setOnLineStatus(false));
    // });
};
