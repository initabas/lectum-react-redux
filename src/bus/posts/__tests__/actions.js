import { postsActions } from '../actions';
// import types from '../types';

describe('posts actions', () => {
    test('fetchPosts should produce a valid action creator', () => {
        expect(postsActions.fetchPosts()).toMatchSnapshot();
    });
    test('fetchPostsSuccess should produce a valid action creator', () => {
        expect(postsActions.fetchPostsSuccess(__.posts)).toMatchSnapshot();
    });
    test('fetchPostsFail should produce a valid action creator', () => {
        expect(postsActions.fetchPostsFail(__.error)).toMatchSnapshot();
    });

    test('createPost should produce a valid action creator', () => {
        expect(postsActions.createPost(__.comment)).toMatchSnapshot();
    });
    test('createPostSuccess should produce a valid action creator', () => {
        expect(postsActions.createPostSuccess(__.post1)).toMatchSnapshot();
    });
    test('createPostFail should produce a valid action creator', () => {
        expect(postsActions.createPostFail(__.error)).toMatchSnapshot();
    });

    test('deletePost should produce a valid action creator', () => {
        expect(postsActions.deletePost(__.postId)).toMatchSnapshot();
    });
    test('deletePostSuccess should produce a valid action creator', () => {
        expect(postsActions.deletePostSuccess(__.postId)).toMatchSnapshot();
    });
    test('deletePostFail should produce a valid action creator', () => {
        expect(postsActions.deletePostFail(__.error)).toMatchSnapshot();
    });

    test('likePost should produce a valid action creator', () => {
        expect(postsActions.likePost(__.postId)).toMatchSnapshot();
    });
    test('likePostSuccess should produce a valid action creator', () => {
        expect(postsActions.likePostSuccess(__.likedPostIds)).toMatchSnapshot();
    });
    test('likePostFail should produce a valid action creator', () => {
        expect(postsActions.likePostFail(__.error)).toMatchSnapshot();
    });

    test('dislikePost should produce a valid action creator', () => {
        expect(postsActions.dislikePost(__.postId)).toMatchSnapshot();
    });
    test('dislikePostSuccess should produce a valid action creator', () => {
        expect(postsActions.dislikePostSuccess(__.likedPostIds)).toMatchSnapshot();
    });
    test('dislikePostFail should produce a valid action creator', () => {
        expect(postsActions.dislikePostFail(__.error)).toMatchSnapshot();
    });
    
});


