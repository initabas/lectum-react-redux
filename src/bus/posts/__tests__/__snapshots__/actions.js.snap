// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`posts actions createPost should produce a valid action creator 1`] = `
Object {
  "payload": "NOT_COMMENT",
  "type": "CREATE_POST",
}
`;

exports[`posts actions createPostFail should produce a valid action creator 1`] = `
Object {
  "error": true,
  "payload": "NOT_AN_ERROR",
  "type": "CREATE_POST_FAIL",
}
`;

exports[`posts actions createPostSuccess should produce a valid action creator 1`] = `
Object {
  "payload": Object {
    "author": Object {
      "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
      "firstName": "Tristian",
      "groupId": "844t1zdqp0",
      "id": "34edd9c052db76aeff2b33es",
      "lastName": "Farrell",
    },
    "comment": "Iusto aut quae non asperiores eius illo ut.",
    "created": 1508760000,
    "id": "59edd93052db76aefs2b36ee_TEST",
    "likes": Array [
      Object {
        "firstName": "Harold",
        "id": "52edd9352db76aeff2b34ed",
        "lastName": "Cassin",
      },
    ],
  },
  "type": "CREATE_POST_SUCCESS",
}
`;

exports[`posts actions deletePost should produce a valid action creator 1`] = `
Object {
  "payload": "NOT_ID",
  "type": "DELETE_POST",
}
`;

exports[`posts actions deletePostFail should produce a valid action creator 1`] = `
Object {
  "error": true,
  "payload": "NOT_AN_ERROR",
  "type": "DELETE_POST_FAIL",
}
`;

exports[`posts actions deletePostSuccess should produce a valid action creator 1`] = `
Object {
  "payload": "NOT_ID",
  "type": "DELETE_POST_SUCCESS",
}
`;

exports[`posts actions dislikePost should produce a valid action creator 1`] = `
Object {
  "payload": "NOT_ID",
  "type": "DISLIKE_POST",
}
`;

exports[`posts actions dislikePostFail should produce a valid action creator 1`] = `
Object {
  "error": true,
  "payload": "NOT_AN_ERROR",
  "type": "DISLIKE_POST_FAIL",
}
`;

exports[`posts actions dislikePostSuccess should produce a valid action creator 1`] = `
Object {
  "payload": Object {
    "postId": "NOT_ID",
    "user": Object {},
  },
  "type": "DISLIKE_POST_SUCCESS",
}
`;

exports[`posts actions fetchPosts should produce a valid action creator 1`] = `
Object {
  "type": "FETCH_POSTS",
}
`;

exports[`posts actions fetchPostsFail should produce a valid action creator 1`] = `
Object {
  "error": true,
  "payload": "NOT_AN_ERROR",
  "type": "FETCH_POSTS_FAIL",
}
`;

exports[`posts actions fetchPostsSuccess should produce a valid action creator 1`] = `
Object {
  "payload": Array [
    Object {
      "author": Object {
        "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
        "firstName": "Tristian",
        "groupId": "844t1zdqp0",
        "id": "34edd9c052db76aeff2b33es",
        "lastName": "Farrell",
      },
      "comment": "Iusto aut quae non asperiores eius illo ut.",
      "created": 1508760000,
      "id": "59edd93052db76aefs2b36ee_TEST",
      "likes": Array [
        Object {
          "firstName": "Harold",
          "id": "52edd9352db76aeff2b34ed",
          "lastName": "Cassin",
        },
      ],
    },
    Object {
      "author": Object {
        "avatar": "https://lab.lectrum.io/redux/api/image/844t1zdqp0/placeholder.jpg",
        "firstName": "Tristian",
        "groupId": "844t1zdqp0",
        "id": "34edd9c052db76aeff2b33es",
        "lastName": "Farrell",
      },
      "comment": "Iusto aut quae non asperiores eius illo ut.",
      "created": 1508760000,
      "id": "59edd93052db76aefs2b36ed_TEST",
      "likes": Array [
        Object {
          "firstName": "Harold",
          "id": "52edd9352db76aeff2b34ed",
          "lastName": "Cassin",
        },
      ],
    },
  ],
  "type": "FETCH_POSTS_SUCCESS",
}
`;

exports[`posts actions likePost should produce a valid action creator 1`] = `
Object {
  "payload": "NOT_ID",
  "type": "LIKE_POST",
}
`;

exports[`posts actions likePostFail should produce a valid action creator 1`] = `
Object {
  "error": true,
  "payload": "NOT_AN_ERROR",
  "type": "LIKE_POST_FAIL",
}
`;

exports[`posts actions likePostSuccess should produce a valid action creator 1`] = `
Object {
  "payload": Object {
    "postId": "NOT_ID",
    "user": Object {},
  },
  "type": "LIKE_POST_SUCCESS",
}
`;
